services/src/test/java/org/keycloak/test/broker/oidc/AbstractOAuth2IdentityProviderTest.java

```diff
@@ -121,10 +132,54 @@ public void getFederatedIdentity_responseUrlLine() {
 		Assert.assertEquals("458rtf", fi.getId());
 	}
 
-	private TestProvider getTested() {
+	@Test
```

Comment:
We require proper functional tests with driving the tests through the actual login pages. Take a look at https://github.com/keycloak/keycloak/blob/master/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/AbstractBrokerTest.java#L94. It uses Selenium/Webdriver to go to the actual login page, click on the IDP button, fill in the form on the IDP, etc..